src/Dialect/Krnl/DialectBuilder.cpp

```diff
@@ -208,37 +240,14 @@ void KrnlBuilder::printf(StringRef msg) const {
 }
 
 void KrnlBuilder::printf(StringRef msg, Value input, Type inputType) const {
-  StringRef format;
-  TypeSwitch<Type>(inputType)
-      .Case<mlir::Float16Type>([&](mlir::Float16Type) { format = "%g\n"; })
-      .Case<mlir::Float32Type>([&](mlir::Float32Type) { format = "%g\n"; })
-      .Case<mlir::Float64Type>([&](mlir::Float64Type) { format = "%g\n"; })
-      .Case<IntegerType>([&](IntegerType type) {
-        switch (type.getWidth()) {
-        case 1:
-        case 8:
-        case 16:
-        case 32:
-          format = type.isUnsigned() ? "%u\n" : "%d\n";
-          break;
-        case 64:
-          format = type.isUnsigned() ? "%llu\n" : "%lld\n";
-          break;
-        }
-      })
-      .Case<IndexType>([&](IndexType) { format = "%lld\n"; })
-      .Case<onnx_mlir::krnl::StringType>(
-          [&](onnx_mlir::krnl::StringType) { format = "%s\n"; })
-      .Case<LLVM::LLVMPointerType>(
-          [&](LLVM::LLVMPointerType) { format = "%s\n"; })
-      .Default([&](Type type) {
-        llvm::errs() << "type: " << type << "\n";
-        llvm_unreachable("Unhandled type");
-      });
-
+  StringRef format = getFormat(inputType);
   std::string concat(msg.str() + format.str());
   StringRef newFormat(concat);
   b.create<KrnlPrintOp>(loc, newFormat, input);
 }
+void KrnlBuilder::printf(Value input, Type inputType) const {
+  StringRef format = getFormat(inputType);
+  b.create<KrnlPrintOp>(loc, format, input);
```

Comment:
Yes, because in some cases the user might want multiple prints on the same line and if "\n" is silently added that wouldn't be possible. The user can always emit a `krnl.print("\n")` to force a newline.